GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -235,10 +237,105 @@ protected override void Execute(GVFSEnlistment enlistment)
                     {
                         this.PruneFoldersOutsideSparse(tracer, enlistment, sparseTable);
                     }
+
+                    if (needToChangeProjection || this.Prune)
+                    {
+                        // Update the last write times of the parents of folders being added/removed
+                        // so that File Explorer will refresh them
+                        UpdateParentFolderLastWriteTimes(tracer, enlistment.WorkingDirectoryBackingRoot, foldersToRemove, foldersToAdd);
+                    }
                 }
             }
         }
 
+        private static void UpdateParentFolderLastWriteTimes(ITracer tracer, string rootPath, List<string> foldersToRemove, List<string> foldersToAdd)
+        {
+            Stopwatch updateTime = Stopwatch.StartNew();
+
+            HashSet<string> foldersToRefresh = new HashSet<string>(GVFSPlatform.Instance.Constants.PathComparer);
+            foreach (string path in foldersToRemove)
+            {
+                AddNonRootParentPathsToSet(foldersToRefresh, path);
+            }
+
+            foreach (string path in foldersToAdd)
+            {
+                AddNonRootParentPathsToSet(foldersToRefresh, path);
+            }
+
+            DateTime refreshTime = DateTime.Now;
+            int foldersRefreshed = 0;
+            int refreshErrors = 0;
+            bool refreshFailed;
+
+            // Always refresh the root
+            if (SetFolderLastWriteTimeIfOnDisk(tracer, rootPath, refreshTime, out refreshFailed))
+            {
+                ++foldersRefreshed;
+            }
+            else if (refreshFailed)
+            {
+                ++refreshErrors;
+            }
+
+            string folderPathPrefix = $"{rootPath}{Path.DirectorySeparatorChar}";

```

Comment:
Is this for perf so we don't have to use `Path.Combine`?

Reply:
Yep it was for performance (although truth be told I did not measure the perf impact of using `Path.Combine` instead).

I figured that since `GVFSDatabase.NormalizePath` had already been run for all the paths we didn't need to add the extra overhead of `Path.Combine`.

If you think `Path.Combine` is preferable let me know and I can try running some performance tests using it instead.

Reply:
Nope this is good.  Thanks!